File: neza-annotation/src/main/java/com/joyy/neza_annotation/method/FlutterMethodChannel.java
package com.joyy.neza_annotation.method;

import com.joyy.neza_annotation.model.ChannelType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author: Jiang Pengyong
 * @date: 2021/8/13 3:18 下午
 * @email: 56002982@qq.com
 * @des: flutter method channel 的注解
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.SOURCE)
public @interface FlutterMethodChannel {
    ChannelType type();

    String channelName();
}

File: neza-annotation/src/main/java/com/joyy/neza_annotation/method/FlutterMethodChannel.kt
package com.joyy.neza_annotation.method

import com.joyy.neza_annotation.model.ChannelType
import java.lang.annotation.RetentionPolicy

/**
 * @author: Jiang Pengyong
 * @date: 2021/8/13 3:18 下午
 * @email: 56002982@qq.com
 * @des: flutter method channel 的注解
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.SOURCE)
annotation class FlutterMethodChannel(
    val type: ChannelType,
    val channelName: String
)